src/hotspot/share/prims/jvmtiThreadState.cpp

```diff
@@ -1041,6 +1041,14 @@ oop JvmtiThreadState::get_thread_oop() {
   return _thread_oop_h.resolve();
 }
 
+void JvmtiThreadState::update_thread_oop_during_vm_start(JavaThread* thread) {
+  assert(thread->threadObj() != nullptr, "santity check");
+  if (thread->jvmti_thread_state()->get_thread_oop() == nullptr) {
+    _thread_oop_h.release(JvmtiExport::jvmti_oop_storage());
```

Comment:
Nit: This line needs a comment explaining we are releasing an `OopHandle` pointing to `null`.

Reply:
I realized that  there is replace method and use it instead of release/update.